DOC Add note about Memory pickle security consideration (#1722)
joblib/joblib: doc/memory.rst

    Running f(2)
    2

> .. note::

Reviewer: Would it be possible to make it explicit that this note is actually a security warning?

```suggestion
.. warning::
```

>   :class:`~joblib.Memory` use pickle to load saved results from files, which
>   means that pickle security considerations apply, see `pickle documentation
>   <https://docs.python.org/3/library/pickle.html>`_ for more details. In
>   particular, only use a cache directory if you trust it.

Reviewer 2: ```suggestion
  :class:`~joblib.Memory` use pickle to load saved results from the file system.
  This means that pickle security considerations apply, see `pickle documentation
  <https://docs.python.org/3/library/pickle.html>`_ for more details.
  In particular, only use a cache directory if you trust it, and if there is a security
  concern about the filesystem, proper safeguards should be instanciated.
```

Author: Can you explain a bit what you mean by "proper safeguards should be instanciated" means :wink:.

Author: OK I removed the ending because I find it a bit too vague.

I have to admit I am not 100% happy with the wording but it's a start ...

>   In particular, only use a cache directory if you trust it.

Reviewer: Let's be more precise in documenting the potential attack vector:

```suggestion
  In particular, if an adversary gains write access to the cache directory,
  it could lead to arbitrary code execution by the process(es) when 
  loading tempered cached values from that folder.
```
